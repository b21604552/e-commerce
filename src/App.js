import './App.css'
import React, { Component } from 'react'
import Container from './main/Container'
import Header from './main/Header'
import { connect } from 'react-redux'

class App extends Component {
  constructor (props) {
    super(props)
    this.state = {}
  }

  componentDidMount () {

  }

  render () {
    return (
      <div className={'main-container'}>
        <Header/>
        <Container/>
      </div>
    )
  }
}

const mapStateToProps = (state, ownProps) => ({
  headerTitle: state.GeneralReducer.headerTitle,
  searchKeyword: state.GeneralReducer.searchKeyword,
  orderValue: state.GeneralReducer.orderValue,
  itemCountInBasket: state.GeneralReducer.itemCountInBasket,
  modalFinished: state.GeneralReducer.modalFinished,
  modalVisible: state.GeneralReducer.modalVisible
})

export default connect(mapStateToProps)(App)
