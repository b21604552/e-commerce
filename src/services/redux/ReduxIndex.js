import { createStore } from 'redux'
import reducerCombined from './reducerCombined'

export const reduxStore = createStore(reducerCombined)
