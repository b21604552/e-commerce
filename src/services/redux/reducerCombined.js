import { combineReducers } from 'redux'
import GeneralReducer from './reducers/GeneralReducer'

const reducerCombined = combineReducers({
  GeneralReducer: GeneralReducer
})

export default reducerCombined
