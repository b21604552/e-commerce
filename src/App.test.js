import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { reduxStore } from './services/redux/ReduxIndex'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer';

describe('App Test', () => {

  let page = (
    <React.StrictMode>
      <Provider store={reduxStore}>
        <App/>
      </Provider>
    </React.StrictMode>
  )

  it('Renders App without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(page, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Renders App correctly', () => {
    const div = renderer.create(page).toJSON();
    expect(div).toMatchSnapshot();
  });
})