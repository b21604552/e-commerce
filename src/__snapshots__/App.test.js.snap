// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test Renders App correctly 1`] = `
<div
  className="main-container"
>
  <div
    className="header-wrapper"
  >
    <div
      className="header"
    >
      <div
        className="logo-wrapper"
      >
        <img
          alt="logo"
          src="logo.svg"
        />
      </div>
      <div
        className="search-wrapper"
      >
        <span
          className="prefix"
        >
          <img
            alt="search-icon"
            src="search-icon.svg"
          />
        </span>
        <input
          className="search-input"
          id="search-input"
          onChange={[Function]}
          placeholder="25 milyon’dan fazla ürün içerisinde ara"
          type="text"
        />
      </div>
      <div
        className="basket-wrapper"
      >
        <div
          className="basket-button"
        >
          <div>
            Sepetim
          </div>
        </div>
        <div
          className="basket-opened"
        >
          <div
            className="basket-empty-wrapper"
          >
            <div
              className="basket-empty"
            >
              Sepetinizde ürün bulunmamaktadır.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="info-wrapper"
  >
    <div
      className="category-wrapper"
    >
      <div
        className="category"
      >
        Hoşgeldiniz
      </div>
      <div
        className="search-keyword-wrapper"
      />
    </div>
    <div
      className="filter-wrapper"
    >
      <select
        className="custom-select"
        onChange={[Function]}
        value=""
      >
        <option
          disabled={true}
          hidden={true}
          value=""
        >
          Sıralama
        </option>
        <option
          value="0"
        >
          En Düşük Fiyat
        </option>
        <option
          value="1"
        >
          En Yüksek Fiyat
        </option>
        <option
          value="2"
        >
          En Yeniler (A&gt;Z)
        </option>
        <option
          value="3"
        >
          En Yeniler (Z&gt;A)
        </option>
      </select>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="side-filter-wrapper"
    >
      <div
        className="side-filter-element"
      >
        <div
          className="filter-element-header"
        >
          Renk
        </div>
        <div
          className="filter-option-wrapper"
        >
          <div
            className="filter-option"
            onClick={[Function]}
          >
            Siyah
             (
            10
            )
          </div>
          <div
            className="filter-option"
            onClick={[Function]}
          >
            Lacivert
             (
            10
            )
          </div>
          <div
            className="filter-option"
            onClick={[Function]}
          >
            Sarı
             (
            10
            )
          </div>
          <div
            className="filter-option"
            onClick={[Function]}
          >
            Beyaz
             (
            10
            )
          </div>
        </div>
      </div>
      <div
        className="side-filter-element"
      >
        <div
          className="filter-element-header"
        >
          Sıralama
        </div>
        <div
          className="filter-option-wrapper"
        >
          <div
            className="filter-option"
            id="order-0"
            onClick={[Function]}
          >
            En Düşük Fiyat
          </div>
          <div
            className="filter-option"
            id="order-1"
            onClick={[Function]}
          >
            En Yüksek Fiyat
          </div>
          <div
            className="filter-option"
            id="order-2"
            onClick={[Function]}
          >
            En Yeniler (A&gt;Z)
          </div>
          <div
            className="filter-option"
            id="order-3"
            onClick={[Function]}
          >
            En Yeniler (Z&gt;A)
          </div>
        </div>
      </div>
      <div
        className="side-filter-element"
      >
        <div
          className="filter-element-header"
        >
          Marka
        </div>
        <div
          className="filter-option-wrapper"
        >
          <div
            className="filter-option"
            onClick={[Function]}
          >
            Apple
             (
            16
            )
          </div>
          <div
            className="filter-option"
            onClick={[Function]}
          >
            Samsung
             (
            12
            )
          </div>
          <div
            className="filter-option"
            onClick={[Function]}
          >
            Xiaomi
             (
            4
            )
          </div>
          <div
            className="filter-option"
            onClick={[Function]}
          >
            Huawei
             (
            4
            )
          </div>
          <div
            className="filter-option"
            onClick={[Function]}
          >
            LG
             (
            4
            )
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        className="products-wrapper"
      >
        <div
          className="product-wrapper"
        >
          <div
            className="product-image-wrapper"
          >
            <img
              alt="product-0"
              src="Apple.png"
            />
          </div>
          <div
            className="product-info-wrapper"
          >
            <div
              className="product-title"
            >
              Apple iPhone 11 32GB
            </div>
            <div
              className="product-properties-info-wrapper"
            >
              <div
                className="product-properties-element"
              >
                <div
                  className="product-properties-label"
                >
                  Marka:
                </div>
                <div
                  className="product-properties-value"
                >
                  Apple
                </div>
              </div>
              <div
                className="product-properties-element"
              >
                <div
                  className="product-properties-label"
                >
                  Renk:
                </div>
                <div
                  className="product-properties-value"
                >
                  Siyah
                </div>
              </div>
            </div>
            <div
              className="product-price-info-wrapper"
            >
              <div
                className="product-price-element"
              >
                109.12
                 TL
              </div>
              <div
                className="product-old-price-element"
              >
                <div
                  className="product-old-price"
                >
                  124.00
                   TL
                </div>
                <div
                  className="product-discount"
                >
                  12
                   %
                </div>
              </div>
            </div>
          </div>
          <div
            className="product-add-cart-wrapper"
          >
            <button
              className="product-button"
              onClick={[Function]}
            >
              Sepete Ekle
            </button>
          </div>
        </div>
        <div
          className="product-wrapper"
        >
          <div
            className="product-image-wrapper"
          >
            <img
              alt="product-1"
              src="Apple.png"
            />
          </div>
          <div
            className="product-info-wrapper"
          >
            <div
              className="product-title"
            >
              Apple iPhone 11 32GB
            </div>
            <div
              className="product-properties-info-wrapper"
            >
              <div
                className="product-properties-element"
              >
                <div
                  className="product-properties-label"
                >
                  Marka:
                </div>
                <div
                  className="product-properties-value"
                >
                  Apple
                </div>
              </div>
              <div
                className="product-properties-element"
              >
                <div
                  className="product-properties-label"
                >
                  Renk:
                </div>
                <div
                  className="product-properties-value"
                >
                  Lacivert
                </div>
              </div>
            </div>
            <div
              className="product-price-info-wrapper"
            >
              <div
                className="product-price-element"
              >
                109.12
                 TL
              </div>
              <div
                className="product-old-price-element"
              >
                <div
                  className="product-old-price"
                >
                  124.00
                   TL
                </div>
                <div
                  className="product-discount"
                >
                  12
                   %
                </div>
              </div>
            </div>
          </div>
          <div
            className="product-add-cart-wrapper"
          >
            <button
              className="product-button"
              onClick={[Function]}
            >
              Sepete Ekle
            </button>
          </div>
        </div>
        <div
          className="product-wrapper"
        >
          <div
            className="product-image-wrapper"
          >
            <img
              alt="product-2"
              src="Apple.png"
            />
          </div>
          <div
            className="product-info-wrapper"
          >
            <div
              className="product-title"
            >
              Apple iPhone 11 32GB
            </div>
            <div
              className="product-properties-info-wrapper"
            >
              <div
                className="product-properties-element"
              >
                <div
                  className="product-properties-label"
                >
                  Marka:
                </div>
                <div
                  className="product-properties-value"
                >
                  Apple
                </div>
              </div>
              <div
                className="product-properties-element"
              >
                <div
                  className="product-properties-label"
                >
                  Renk:
                </div>
                <div
                  className="product-properties-value"
                >
                  Sarı
                </div>
              </div>
            </div>
            <div
              className="product-price-info-wrapper"
            >
              <div
                className="product-price-element"
              >
                109.12
                 TL
              </div>
              <div
                className="product-old-price-element"
              >
                <div
                  className="product-old-price"
                >
                  124.00
                   TL
                </div>
                <div
                  className="product-discount"
                >
                  12
                   %
                </div>
              </div>
            </div>
          </div>
          <div
            className="product-add-cart-wrapper"
          >
            <button
              className="product-button"
              onClick={[Function]}
            >
              Sepete Ekle
            </button>
          </div>
        </div>
        <div
          className="product-wrapper"
        >
          <div
            className="product-image-wrapper"
          >
            <img
              alt="product-3"
              src="Apple.png"
            />
          </div>
          <div
            className="product-info-wrapper"
          >
            <div
              className="product-title"
            >
              Apple iPhone 11 32GB
            </div>
            <div
              className="product-properties-info-wrapper"
            >
              <div
                className="product-properties-element"
              >
                <div
                  className="product-properties-label"
                >
                  Marka:
                </div>
                <div
                  className="product-properties-value"
                >
                  Apple
                </div>
              </div>
              <div
                className="product-properties-element"
              >
                <div
                  className="product-properties-label"
                >
                  Renk:
                </div>
                <div
                  className="product-properties-value"
                >
                  Beyaz
                </div>
              </div>
            </div>
            <div
              className="product-price-info-wrapper"
            >
              <div
                className="product-price-element"
              >
                109.12
                 TL
              </div>
              <div
                className="product-old-price-element"
              >
                <div
                  className="product-old-price"
                >
                  124.00
                   TL
                </div>
                <div
                  className="product-discount"
                >
                  12
                   %
                </div>
              </div>
            </div>
          </div>
          <div
            className="product-add-cart-wrapper"
          >
            <button
              className="product-button"
              onClick={[Function]}
            >
              Sepete Ekle
            </button>
          </div>
        </div>
        <div
          className="product-wrapper"
        >
          <div
            className="product-image-wrapper"
          >
            <img
              alt="product-4"
              src="Apple.png"
            />
          </div>
          <div
            className="product-info-wrapper"
          >
            <div
              className="product-title"
            >
              Apple iPhone 11 64GB
            </div>
            <div
              className="product-properties-info-wrapper"
            >
              <div
                className="product-properties-element"
              >
                <div
                  className="product-properties-label"
                >
                  Marka:
                </div>
                <div
                  className="product-properties-value"
                >
                  Apple
                </div>
              </div>
              <div
                className="product-properties-element"
              >
                <div
                  className="product-properties-label"
                >
                  Renk:
                </div>
                <div
                  className="product-properties-value"
                >
                  Siyah
                </div>
              </div>
            </div>
            <div
              className="product-price-info-wrapper"
            >
              <div
                className="product-price-element"
              >
                194.88
                 TL
              </div>
              <div
                className="product-old-price-element"
              >
                <div
                  className="product-old-price"
                >
                  224.00
                   TL
                </div>
                <div
                  className="product-discount"
                >
                  13
                   %
                </div>
              </div>
            </div>
          </div>
          <div
            className="product-add-cart-wrapper"
          >
            <button
              className="product-button"
              onClick={[Function]}
            >
              Sepete Ekle
            </button>
          </div>
        </div>
        <div
          className="product-wrapper"
        >
          <div
            className="product-image-wrapper"
          >
            <img
              alt="product-5"
              src="Apple.png"
            />
          </div>
          <div
            className="product-info-wrapper"
          >
            <div
              className="product-title"
            >
              Apple iPhone 11 64GB
            </div>
            <div
              className="product-properties-info-wrapper"
            >
              <div
                className="product-properties-element"
              >
                <div
                  className="product-properties-label"
                >
                  Marka:
                </div>
                <div
                  className="product-properties-value"
                >
                  Apple
                </div>
              </div>
              <div
                className="product-properties-element"
              >
                <div
                  className="product-properties-label"
                >
                  Renk:
                </div>
                <div
                  className="product-properties-value"
                >
                  Lacivert
                </div>
              </div>
            </div>
            <div
              className="product-price-info-wrapper"
            >
              <div
                className="product-price-element"
              >
                194.88
                 TL
              </div>
              <div
                className="product-old-price-element"
              >
                <div
                  className="product-old-price"
                >
                  224.00
                   TL
                </div>
                <div
                  className="product-discount"
                >
                  13
                   %
                </div>
              </div>
            </div>
          </div>
          <div
            className="product-add-cart-wrapper"
          >
            <button
              className="product-button"
              onClick={[Function]}
            >
              Sepete Ekle
            </button>
          </div>
        </div>
        <div
          className="product-wrapper"
        >
          <div
            className="product-image-wrapper"
          >
            <img
              alt="product-6"
              src="Apple.png"
            />
          </div>
          <div
            className="product-info-wrapper"
          >
            <div
              className="product-title"
            >
              Apple iPhone 11 64GB
            </div>
            <div
              className="product-properties-info-wrapper"
            >
              <div
                className="product-properties-element"
              >
                <div
                  className="product-properties-label"
                >
                  Marka:
                </div>
                <div
                  className="product-properties-value"
                >
                  Apple
                </div>
              </div>
              <div
                className="product-properties-element"
              >
                <div
                  className="product-properties-label"
                >
                  Renk:
                </div>
                <div
                  className="product-properties-value"
                >
                  Sarı
                </div>
              </div>
            </div>
            <div
              className="product-price-info-wrapper"
            >
              <div
                className="product-price-element"
              >
                194.88
                 TL
              </div>
              <div
                className="product-old-price-element"
              >
                <div
                  className="product-old-price"
                >
                  224.00
                   TL
                </div>
                <div
                  className="product-discount"
                >
                  13
                   %
                </div>
              </div>
            </div>
          </div>
          <div
            className="product-add-cart-wrapper"
          >
            <button
              className="product-button"
              onClick={[Function]}
            >
              Sepete Ekle
            </button>
          </div>
        </div>
        <div
          className="product-wrapper"
        >
          <div
            className="product-image-wrapper"
          >
            <img
              alt="product-7"
              src="Apple.png"
            />
          </div>
          <div
            className="product-info-wrapper"
          >
            <div
              className="product-title"
            >
              Apple iPhone 11 64GB
            </div>
            <div
              className="product-properties-info-wrapper"
            >
              <div
                className="product-properties-element"
              >
                <div
                  className="product-properties-label"
                >
                  Marka:
                </div>
                <div
                  className="product-properties-value"
                >
                  Apple
                </div>
              </div>
              <div
                className="product-properties-element"
              >
                <div
                  className="product-properties-label"
                >
                  Renk:
                </div>
                <div
                  className="product-properties-value"
                >
                  Beyaz
                </div>
              </div>
            </div>
            <div
              className="product-price-info-wrapper"
            >
              <div
                className="product-price-element"
              >
                194.88
                 TL
              </div>
              <div
                className="product-old-price-element"
              >
                <div
                  className="product-old-price"
                >
                  224.00
                   TL
                </div>
                <div
                  className="product-discount"
                >
                  13
                   %
                </div>
              </div>
            </div>
          </div>
          <div
            className="product-add-cart-wrapper"
          >
            <button
              className="product-button"
              onClick={[Function]}
            >
              Sepete Ekle
            </button>
          </div>
        </div>
        <div
          className="product-wrapper"
        >
          <div
            className="product-image-wrapper"
          >
            <img
              alt="product-8"
              src="Apple.png"
            />
          </div>
          <div
            className="product-info-wrapper"
          >
            <div
              className="product-title"
            >
              Apple iPhone 11 128GB
            </div>
            <div
              className="product-properties-info-wrapper"
            >
              <div
                className="product-properties-element"
              >
                <div
                  className="product-properties-label"
                >
                  Marka:
                </div>
                <div
                  className="product-properties-value"
                >
                  Apple
                </div>
              </div>
              <div
                className="product-properties-element"
              >
                <div
                  className="product-properties-label"
                >
                  Renk:
                </div>
                <div
                  className="product-properties-value"
                >
                  Siyah
                </div>
              </div>
            </div>
            <div
              className="product-price-info-wrapper"
            >
              <div
                className="product-price-element"
              >
                278.64
                 TL
              </div>
              <div
                className="product-old-price-element"
              >
                <div
                  className="product-old-price"
                >
                  324.00
                   TL
                </div>
                <div
                  className="product-discount"
                >
                  14
                   %
                </div>
              </div>
            </div>
          </div>
          <div
            className="product-add-cart-wrapper"
          >
            <button
              className="product-button"
              onClick={[Function]}
            >
              Sepete Ekle
            </button>
          </div>
        </div>
        <div
          className="product-wrapper"
        >
          <div
            className="product-image-wrapper"
          >
            <img
              alt="product-9"
              src="Apple.png"
            />
          </div>
          <div
            className="product-info-wrapper"
          >
            <div
              className="product-title"
            >
              Apple iPhone 11 128GB
            </div>
            <div
              className="product-properties-info-wrapper"
            >
              <div
                className="product-properties-element"
              >
                <div
                  className="product-properties-label"
                >
                  Marka:
                </div>
                <div
                  className="product-properties-value"
                >
                  Apple
                </div>
              </div>
              <div
                className="product-properties-element"
              >
                <div
                  className="product-properties-label"
                >
                  Renk:
                </div>
                <div
                  className="product-properties-value"
                >
                  Lacivert
                </div>
              </div>
            </div>
            <div
              className="product-price-info-wrapper"
            >
              <div
                className="product-price-element"
              >
                278.64
                 TL
              </div>
              <div
                className="product-old-price-element"
              >
                <div
                  className="product-old-price"
                >
                  324.00
                   TL
                </div>
                <div
                  className="product-discount"
                >
                  14
                   %
                </div>
              </div>
            </div>
          </div>
          <div
            className="product-add-cart-wrapper"
          >
            <button
              className="product-button"
              onClick={[Function]}
            >
              Sepete Ekle
            </button>
          </div>
        </div>
        <div
          className="product-wrapper"
        >
          <div
            className="product-image-wrapper"
          >
            <img
              alt="product-10"
              src="Apple.png"
            />
          </div>
          <div
            className="product-info-wrapper"
          >
            <div
              className="product-title"
            >
              Apple iPhone 11 128GB
            </div>
            <div
              className="product-properties-info-wrapper"
            >
              <div
                className="product-properties-element"
              >
                <div
                  className="product-properties-label"
                >
                  Marka:
                </div>
                <div
                  className="product-properties-value"
                >
                  Apple
                </div>
              </div>
              <div
                className="product-properties-element"
              >
                <div
                  className="product-properties-label"
                >
                  Renk:
                </div>
                <div
                  className="product-properties-value"
                >
                  Sarı
                </div>
              </div>
            </div>
            <div
              className="product-price-info-wrapper"
            >
              <div
                className="product-price-element"
              >
                278.64
                 TL
              </div>
              <div
                className="product-old-price-element"
              >
                <div
                  className="product-old-price"
                >
                  324.00
                   TL
                </div>
                <div
                  className="product-discount"
                >
                  14
                   %
                </div>
              </div>
            </div>
          </div>
          <div
            className="product-add-cart-wrapper"
          >
            <button
              className="product-button"
              onClick={[Function]}
            >
              Sepete Ekle
            </button>
          </div>
        </div>
        <div
          className="product-wrapper"
        >
          <div
            className="product-image-wrapper"
          >
            <img
              alt="product-11"
              src="Apple.png"
            />
          </div>
          <div
            className="product-info-wrapper"
          >
            <div
              className="product-title"
            >
              Apple iPhone 11 128GB
            </div>
            <div
              className="product-properties-info-wrapper"
            >
              <div
                className="product-properties-element"
              >
                <div
                  className="product-properties-label"
                >
                  Marka:
                </div>
                <div
                  className="product-properties-value"
                >
                  Apple
                </div>
              </div>
              <div
                className="product-properties-element"
              >
                <div
                  className="product-properties-label"
                >
                  Renk:
                </div>
                <div
                  className="product-properties-value"
                >
                  Beyaz
                </div>
              </div>
            </div>
            <div
              className="product-price-info-wrapper"
            >
              <div
                className="product-price-element"
              >
                278.64
                 TL
              </div>
              <div
                className="product-old-price-element"
              >
                <div
                  className="product-old-price"
                >
                  324.00
                   TL
                </div>
                <div
                  className="product-discount"
                >
                  14
                   %
                </div>
              </div>
            </div>
          </div>
          <div
            className="product-add-cart-wrapper"
          >
            <button
              className="product-button"
              onClick={[Function]}
            >
              Sepete Ekle
            </button>
          </div>
        </div>
      </div>
      <div
        className="pagination-wrapper"
      >
        <div
          className="pagination-element pagination-element-active"
          id="page-1"
          onClick={[Function]}
        >
          1
        </div>
        <div
          className="pagination-element"
          id="page-2"
          onClick={[Function]}
        >
          2
        </div>
        <div
          className="pagination-element"
          id="page-3"
          onClick={[Function]}
        >
          3
        </div>
        <div
          className="pagination-element"
          id="page-4"
          onClick={[Function]}
        >
          4
        </div>
        <div
          className="pagination-element pagination-next"
          id="next-page"
          onClick={[Function]}
        >
          &gt;
        </div>
      </div>
    </div>
  </div>
</div>
`;
